### DISCUSSION TITLE
Group permissions for model not persistent

### DISCUSSION THREAD

**Discussion by Ivar-Nelson** [open-webui/open-webui#8008] [Issues]
# Bug Report

## Important Notes

- **Before submitting a bug report**: Please check the Issues or Discussions section to see if a similar issue or feature request has already been posted. It's likely we're already tracking it! If you’re unsure, start a discussion post first. This will help us efficiently focus on improving the project.

- **Collaborate respectfully**: We value a constructive attitude, so please be mindful of your communication. If negativity is part of your approach, our capacity to engage may be limited. We’re here to help if you’re open to learning and communicating positively. Remember, Open WebUI is a volunteer-driven project managed by a single maintainer and supported by contributors who also have full-time jobs. We appreciate your time and ask that you respect ours.

- **Contributing**: If you encounter an issue, we highly encourage you to submit a pull request or fork the project. We actively work to prevent contributor burnout to maintain the quality and continuity of Open WebUI.

- **Bug reproducibility**: If a bug cannot be reproduced with a `:main` or `:dev` Docker setup, or a pip install with Python 3.11, it may require additional help from the community. In such cases, we will move it to the "issues" Discussions section due to our limited resources. We encourage the community to assist with these issues. Remember, it’s not that the issue doesn’t exist; we need your help!

Note: Please remove the notes above when submitting your post. Thank you for your understanding and support!

---

## Installation Method

[Describe the method you used to install the project, e.g., git clone, Docker, pip, etc.]

## Environment

- **Open WebUI Version:** [e.g., v0.3.11]
- **Ollama (if applicable):** [e.g., v0.2.0, v0.1.32-rc1]

- **Operating System:** [e.g., Windows 10, macOS Big Sur, Ubuntu 20.04]
- **Browser (if applicable):** [e.g., Chrome 100.0, Firefox 98.0]

**Confirmation:**

- [x] I have read and followed all the instructions provided in the README.md.
- [x] I am on the latest version of both Open WebUI and Ollama.
- [ ] I have included the browser console logs.
- [x] I have included the Docker container logs.
- [ ] I have provided the exact steps to reproduce the bug in the "Steps to Reproduce" section below.

## Expected Behavior:

[Describe what you expected to happen.]

## Actual Behavior:

[Describe what actually happened.]

## Description

**Bug Summary:**
Trying to set the group permissions for a model does not work
Having it private works well for admin.
Making it public also works

Selecting a group for a model works but when saving the new setting it  says update sucess 'model updated successfully' but if entering the settings page no update has been made.

The behaviour is the same regardless of  BYPASSS_MODEL_ACCES_CONTROL

## Reproduction Details

**Steps to Reproduce:**
No special steps taken.

OPENAI_API_KEY="secret"
CUSTOM_NAME="custom name"
ENABLE_SIGNUP="false"
BYPASS_MODEL_ACCESS_CONTROL="true"

## Logs and Screenshots

**Browser Console Logs:**
[Include relevant browser console logs, if applicable]

**Docker Container Logs:**
INFO  [open_webui.apps.openai.main] get_all_models()
INFO  [open_webui.apps.ollama.main] get_all_models()
INFO:     172.22.0.3:48812 - "GET /api/models/base HTTP/1.0" 200 OK
INFO:     172.22.0.3:33178 - "GET /api/v1/tools/ HTTP/1.0" 200 OK
INFO:     172.22.0.3:33182 - "GET /api/v1/functions/ HTTP/1.0" 200 OK
INFO:     172.22.0.3:33188 - "GET /api/v1/knowledge/ HTTP/1.0" 200 OK
INFO:     172.22.0.3:33196 - "GET /api/v1/groups/ HTTP/1.0" 200 OK
INFO:     172.22.0.3:33206 - "POST /api/v1/models/model/update?id=chatgpt-4o-latest HTTP/1.0" 200 OK
INFO  [open_webui.apps.openai.main] get_all_models()
INFO  [open_webui.apps.ollama.main] get_all_models()
INFO:     172.22.0.3:33212 - "GET /api/models HTTP/1.0" 200 OK
INFO:     172.22.0.3:40346 - "GET /api/v1/models/base HTTP/1.0" 200 OK
INFO:     172.22.0.3:40348 - "GET /api/models/base HTTP/1.0" 200 OK
INFO:     172.22.0.3:40362 - "GET /api/v1/tools/ HTTP/1.0" 200 OK
INFO:     172.22.0.3:40374 - "GET /api/v1/functions/ HTTP/1.0" 200 OK
INFO:     172.22.0.3:40384 - "GET /api/v1/knowledge/ HTTP/1.0" 200 OK
INFO:     172.22.0.3:40390 - "GET /api/v1/groups/ HTTP/1.0" 200 OK
INFO:     172.22.0.3:50470 - "POST /api/v1/models/model/update?id=chatgpt-4o-latest HTTP/1.0" 200 OK
INFO  [open_webui.apps.openai.main] get_all_models()
INFO  [open_webui.apps.ollama.main] get_all_models()
INFO:     172.22.0.3:50478 - "GET /api/models HTTP/1.0" 200 OK
INFO:     172.22.0.3:45262 - "GET /api/v1/models/base HTTP/1.0" 200 OK
INFO:     172.22.0.3:45278 - "GET /api/models/base HTTP/1.0" 200 OK
INFO:     172.22.0.3:45294 - "GET /api/v1/tools/ HTTP/1.0" 200 OK
INFO:     172.22.0.3:45302 - "GET /api/v1/functions/ HTTP/1.0" 200 OK
INFO:     172.22.0.3:45310 - "GET /api/v1/knowledge/ HTTP/1.0" 200 OK
INFO:     172.22.0.3:45312 - "GET /api/v1/groups/ HTTP/1.0" 200 OK

**Screenshots/Screen Recordings (if applicable):**
<img width="1120" alt="Screenshot 2024-12-22 at 16 00 46" src="https://github.com/user-attachments/assets/f656231a-4aec-41fc-b1fd-bc42a22f3119" />

## Additional Information

Perhaps the entire settings page is in the process of a makeover. It is somewhat convoluted and there are no "batch" update options.
With that being said, I am very grateful for the project.

## Note

If the bug report is incomplete or does not follow the provided instructions, it may not be addressed. Please ensure that you have followed the steps outlined in the README.md and troubleshooting.md documents, and provide all necessary information for us to reproduce and address the issue. Thank you!

**Comment by MasiGit** [comment]
Same issue, Alerted as "Model Updated Successfully" but the group is still not persisted

**Reply by nandanmv** [reply]
Same issue.

**Reply by zpetkovic** [reply]
Yes, the same. 0.5.4 and 0.5.3

Edit: seems to go away with reboot

**Comment by veleinfobip** [comment]
Description: 
There appears to be an issue with group permissions in the Open WebUI. The interface displays groups correctly after they are added to a model, but users belonging to these groups are unable to access the model as expected. 

![Screenshot 2025-01-17 141503](https://github.com/user-attachments/assets/8dc8b1a8-41ee-48c9-8eb2-6cb840fc1aab)


Steps to Reproduce:  

    Log in as an admin user.
    Create a Model as Admin:  Create a new model within the Open WebUI.
    Select a Group:  Navigate to the group selection section and choose the desired group.
    Add Group to Model:  Click the "+" button to add the selected group to the model.
    Save Changes:  Save the model after adding the group.
    Verify Access:  Attempt to access the model using a user account that belongs to the added group.
     
Expected Behavior: 
After adding a group to a model and saving, the groups should be visible on the model, and all users who are members of these groups should have access to the model. 

Actual Behavior: 
While the groups are displayed on the model after saving, users who are part of the defined groups are still unable to access the model.

**Comment by carlo088** [comment]
any updates on this one? still having same issue with v 0.5.4
